src/main/java/net/fabricmc/loader/gui/FabricStatusTree.java

```diff
@@ -0,0 +1,310 @@
+		CLICK_ONCE;
+	}
+
+	/** No icon is displayed. */
+	public static final String ICON_TYPE_DEFAULT = "";
+
+	/** Generic folder */
+	public static final String ICON_TYPE_FOLDER = "folder";
+
+	/** Generic (unknown contents) file */
+	public static final String ICON_TYPE_UNKNOWN_FILE = "file";
+
+	/** Generic non-fabric jar file. */
+	public static final String ICON_TYPE_JAR_FILE = "jar";
+
+	/** Generic fabric-related jar file. */
+	public static final String ICON_TYPE_FABRIC_JAR_FILE = "jar+fabric";
+
+	/** Something related to fabric. (It's not defined what exactly this is for, but it uses the main fabric logo). */
+	public static final String ICON_TYPE_FABRIC = "fabric";
+
+	/** Generic json file */
+	public static final String ICON_TYPE_JSON = "json";
+
+	/** A file called "fabric.mod.json". */
+	public static final String ICON_TYPE_FABRIC_JSON = "json+fabric";
+
+	/** Java bytecode class file. */
+	public static final String ICON_TYPE_JAVA_CLASS = "java_class";
+
+	/** A folder inside of a java jar. */
+	public static final String ICON_TYPE_PACKAGE = "package";
+
+	/** A folder that contains java class files. */
+	public static final String ICON_TYPE_JAVA_PACKAGE = "java_package";
+
+	/** A tick symbol, used to indicate that something matched. */
+	public static final String ICON_TYPE_TICK = "tick";
+
+	/** A cross symbol, used to indicate that something didn't match. (Although it's not an error). Used as the opposite
+	 * of {@link #ICON_TYPE_TICK} */
+	public static final String ICON_TYPE_LESSER_CROSS = "lesser_cross";
+
+	public final List<FabricStatusTab> tabs = new ArrayList<>();
+	public final List<FabricStatusButton> buttons = new ArrayList<>();
+
+	public String mainText = null;
+
+	public FabricStatusTab addTab(String name) {
+		FabricStatusTab tab = new FabricStatusTab(name);
+		tabs.add(tab);
+		return tab;
+	}
+
+	public FabricStatusButton addButton(String text) {
+		FabricStatusButton button = new FabricStatusButton(text);
+		buttons.add(button);
+		return button;
+	}
+
+	public static final class FabricStatusButton {
+
+		public final String text;
+		public boolean shouldClose, shouldContinue;
+
+		public FabricStatusButton(String text) {
+			this.text = text;
+		}
+
+		public FabricStatusButton makeClose() {
+			shouldClose = true;
+			return this;
+		}
+
+		public FabricStatusButton makeContinue() {
+			this.shouldContinue = true;
+			return this;
+		}
+	}
+
+	public static final class FabricStatusTab {
+		public final FabricStatusNode node;
+
+		/** The minimum warning level to display for this tab. */
+		public FabricTreeWarningLevel filterLevel = FabricTreeWarningLevel.NONE;
+
+		public FabricStatusTab(String name) {
+			this.node = new FabricStatusNode(null, name);
+		}
+
+		public FabricStatusNode addChild(String name) {
+			return node.addChild(name);
+		}
+	}
+
+	public static final class FabricStatusNode {
+
+		private transient FabricStatusNode parent;
```

Comment:
It was sent through gson to a new process for the information gui - however as that requirement was dropped (and you don't want to depend on gson *anyway*) it can be removed.